### PR TITLE
Create docs page for workspace setup

### LINKED ISSUE #76: Create docs page Sailbot Workspace/Setup
### Purpose
Move setup instructions to docs site, as they are quite long and vary depending on OS

- Work on once UBCSailbot/sailbot_workspace#34 is merged

### Page Tree
```yml
nav:
  ...
  - Sailbot Workspace:
    ...
    - Setup: workspace/setup.md
```

### Resources
- https://github.com/UBCSailbot/sailbot_workspace/blob/main/README.md#setup
- @Tsuchijo's notes for getting it working on Arch Linux
    - Run These Two Commands Locally to setup X11 fowarding 
        - Install xhost using `sudo pacman -S xorg-xhost`
        - `export DISPLAY=0.0`
        - `xhost +local:docker`
    - Dev-Containers only works on the right flavor of Vscode, make sure you are using the official Microsoft distribution
        - The one from Pacman doesnt work with microsoft plugins
    - After the image is running on your machine, follow standard procedures
    - [Put this in a collapsible admonition titled Debugging with the bug icon; applies to all non-debian distros I think] To run Ros1, run these commands as the system has trouble resolving the local hostname 
        - `export ROS_HOSTNAME=localhost`
        - `export ROS_MASTER_URI=http://localhost:11311`

### PR DESCRIPTION
**Pull request by eomielan**
This resolves #76

### CODE REVIEW

File: docs/workspace/setup.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        ### Zsh Configuration
+
+        Execute these commands in your terminal:
+
+        ```bash
+        echo 'export MAC_DOCKER_LOCALHOST="docker.for.mac.host.internal"' >> ~/.zshrc
+        echo 'export DISPLAY=:0' >> ~/.zshrc
+        ```
+        If VS Code is open, then restart it.
+
```
Comment (DFriend01) [inline docs/workspace/setup.md]: @patrick-5546 I modified the instructions slightly here. These are probably equivalent, but please check anyways.

**Review by DFriend01 — APPROVED** [review]
Overall, very good work. I did some slight reformatting and proofreading in some places. I also merged a new linter, but I saw that you already addressed the links :smile:

File: mkdocs.yml [hunk 2]
```diff
@@ -123,3 +123,6 @@ nav:
       - Types of Turns: reference/sailing/turning.md
       - Points of Sail: reference/sailing/points_of_sail.md
       - Miscellaneous: reference/sailing/miscellaneous.md
+  - Sailbot Workspace:
+    - workspace/index.md
+    - Setup: workspace/setup.md
```
Comment (DFriend01) [inline mkdocs.yml]: Once we create a new project tab, the workspace pages should be moved under the new project information.

Comment (patrick-5546) [reply mkdocs.yml]: I have created the new project tab in this PR

File: docs/workspace/setup.md (hunk 1 above)
Comment (patrick-5546) [reply docs/workspace/setup.md]: Looks good, will review more in depth later

**Review by patrick-5546 — CHANGES_REQUESTED** [review]
My objective with this PR is completely replace the README in sailbot_workspace. The first step would be to update it with the latest README updates: https://github.com/UBCSailbot/sailbot_workspace/commit/d482f0e76f2a30bac51742bb64cb171d5794893d. Then I will review the PR more in depth. Once this is merged in, link to the overview page in the sailbot_workspace details and delete its README

image

**Review by patrick-5546 — APPROVED** [review]
I updated the setup and run instructions and took advantage of the capabilities of mkdocs material to make them more readable. Feel free to build on top of these changes.

**Comment by DFriend01** [comment]
@eomielan You can merge when you are ready

**Review by DFriend01 — APPROVED** [review]

**Pull request merged**